### DISCUSSION TITLE
Behavior change in 4.6.0 for async method using ref struct not spanning await

### DISCUSSION THREAD

**Discussion by RobinsonWM** [dotnet/roslyn#67317] [Q&A] [answered]
I have some C# code that has compiled in the past but began producing a compiler error after I updated Roslyn from 4.5.0 to 4.6.0 (by its inclusion in Visual Studio 2022 17.6.0 preview 2.0.)  I'm trying to discern whether this change in behavior was intentional (i.e. fixing an old compiler bug) or a compiler regression.

Specifically, prior to 4.6.0, it would compile if I used a `ref struct` in a `using` statement in an `async` method, as long as the `using` block did not contain any `await`s:
```csharp
using System;
using System.Threading.Tasks;

public class C {
    public async Task M() {
        await Task.Delay(1);
        using (new S()) {
            Console.WriteLine("No await in here!");
        }
        await Task.Delay(1);
    }
}

public ref struct S {
    public void Dispose() {}
}
```

Result with `4.5.0-6.23127.3 (e2bc27d2)` and earlier: the code compiles
Result with `4.6.0-2.23121.6 (e728efff)`: `error CS9104: A using statement resource of type 'S' cannot be used in async methods or async lambda expressions.`

**Comment by jcouv** [comment, marked as answer]
This is intentional and due to https://github.com/dotnet/roslyn/pull/66264
But somehow I missed adding an entry to the compat break document. Will add that.
